sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/HDFSBackedStateStoreProvider.scala

```diff
@@ -285,6 +317,33 @@ private[sql] class HDFSBackedStateStoreProvider extends StateStoreProvider with
     }
   }
 
+  /**
+   * Consturct the state at endVersion from snapshot from startVersion.
+   * Returns a new [[HDFSBackedStateStoreMap]]
+   * @param startVersion checkpoint version of the snapshot to start with
```

Comment:
Ideally the semantic (intention) needs to be explained in either method name or param name. The behavior isn't clear until someone reads the method doc. If we address the name (either method or param), I guess method doc is not going to be needed (given this is a private method).

Reply:
same here